config/synapse.conf.json

```diff
@@ -54,9 +54,12 @@
   },
   "haproxy": {
     "reload_command": "sudo service haproxy reload",
+    "check_command": "haproxy -c -f /etc/haproxy/haproxy-staging.cfg",
+    "staging_config_file_path": "/etc/haproxy/haproxy-staging.cfg",
     "config_file_path": "/etc/haproxy/haproxy.cfg",
     "socket_file_path": "/var/haproxy/stats.sock",
     "do_writes": false,
+    "do_checks": false,
```

Comment:
these 3 options we are exposing should be grouped together? maybe like this?

"do_check_config": {
       "tmp_file_path": "/etc/haproxy/haproxy.tmp.cfg",
        "check_command": "haproxy -c -f /etc/haproxy/haproxy.tmp.cfg"
}